ambry-server/src/test/java/com.github.ambry.server/AmbryRequestsTest.java

```diff
@@ -518,6 +519,22 @@ public void listOfOriginalStoreKeysGetTest() throws Exception {
     }
     sendAndVerifyGetOriginalStoreKeys(blobIds, ServerErrorCode.No_Error);
 
+    // test with duplicates
+    List<BlobId> blobIdsWithDups = new ArrayList<>(blobIds);
+    // add the same blob ids
+    blobIdsWithDups.addAll(blobIds);
+    // add converted ids
+    conversionMap.values().forEach(id -> blobIdsWithDups.add((BlobId) id));
+    sendAndVerifyGetOriginalStoreKeys(blobIdsWithDups, ServerErrorCode.No_Error);
+    // store must not have received duplicates
+    assertEquals("Size is not as expected", blobIds.size(), MockStorageManager.idsReceived.size());
+    for (int i = 0; i < blobIds.size(); i++) {
+      BlobId key = blobIds.get(i);
+      StoreKey converted = conversionMap.get(key);
+      assertEquals(key + "/" + converted + " was not received at the store", converted,
+          MockStorageManager.idsReceived.get(i));
+    }
+
     // Check a valid key mapped to null
```

Comment:
Minor: In initial code line 547 ~ line 533 could be changed as follows (maybe more concise):
```
// Check a invalid key mapped to null
validKeysInStore.remove(originalBlobId);
sendAndVerifyGetOriginalStoreKeys(blobIds, ServerErrorCode.Blob_Not_Found);
```

Reply:
That's true. Since it is unrelated however, I will hold off on the change